examples/vector_databases/supabase/semantic-search.mdx

```diff
@@ -0,0 +1,276 @@
+# Semantic search using Supabase Vector
+
+The purpose of this guide is to demonstrate how to store OpenAI embeddings in [Supabase Vector](https://supabase.com/docs/guides/ai) (Postgres + pgvector) for the purposes of semantic search.
+
+[Supabase](https://supabase.com/docs) is an open-source Firebase alternative built on top of [Postgres](https://en.wikipedia.org/wiki/PostgreSQL), a production-grade SQL database. Since Supabase Vector is built on [pgvector](https://github.com/pgvector/pgvector), you can store your embeddings within the same database that holds the rest of your application data. When combined with pgvector's indexing algorithms, vector search remains [fast at large scales](https://supabase.com/blog/increase-performance-pgvector-hnsw).
+
+Supabase adds an ecosystem of services and tools to make app development as quick as possible (such as an [auto-generated REST API](https://postgrest.org/)). We'll use these services to store and query embeddings within Postgres.
+
+This guide covers:
+
+1. [Setting up your database](#setup-database)
+1. [Creating a SQL table](#create-a-vector-table) that can store vector data
+1. [Generating OpenAI embeddings](#generate-openai-embeddings) using OpenAI's JavaScript client
+1. [Storing the embeddings](#store-embeddings-in-database) in your SQL table using the Supabase JavaScript client
+1. [Performing semantic search](#semantic-search) over the embeddings using a Postgres function and the Supabase JavaScript client
+
+## Setup database
+
+First head over to https://database.new to provision your Supabase database. This will create a Postgres database on the Supabase cloud platform. Alternatively, you can follow the [local development](https://supabase.com/docs/guides/cli/getting-started) options if you prefer to run your database locally using Docker.
+
+In the studio, jump to the [SQL editor](https://supabase.com/dashboard/project/_/sql/new) and execute the following SQL to enable pgvector:
+
+```sql
+-- Enable the pgvector extension
+create extension if not exists vector;
+```
+
+> In a production application, best practice is to use [database migrations](https://supabase.com/docs/guides/cli/local-development#database-migrations) so that all SQL operations are managed within source control. To keep things simple in this guide, we'll execute queries directly in the SQL Editor. If you are building a production app, feel free to move these into a database migration.
```

Comment:
```suggestion
> In a production application, the best practice is to use [database migrations](https://supabase.com/docs/guides/cli/local-development#database-migrations) so that all SQL operations are managed within source control. To keep things simple in this guide, we'll execute queries directly in the SQL Editor. If you are building a production app, feel free to move these into a database migration.
```